### PR TITLE
Feature IM-08-08 (Einschulungsformulare drucken)

### PR DESCRIPTION
**Pull request by lenav99**
Schaut mal ob das so alles passt. Danke

### CODE REVIEW

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 1]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
     @Override
     public void initElements() throws DocumentException {
-        Paragraph paragraphEnrollmentDate = new Paragraph("Die Einschulung findet am " + enrollmentDate.format(dateFormatter) + " um " + enrollmentDate.format(timeFormatter) + " Uhr statt.", FONT_TITLE);
-        
-        paragraphEnrollmentDate.setAlignment(Paragraph.ALIGN_CENTER);
+
+        Paragraph paragraphEnrollmentDate = createEnrollmentDateParagraph();
+        PdfPTable tableStudent = createStudentTable();
+        PdfPTable tableEducation = createEducationTable();
+        PdfPTable tableFirm = createFirmTable();
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:104]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'tableFirm' declaration and its first usage is 9, but allowed 8.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 2]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        PdfPTable tableContact = createContactTable();
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:105]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'tableContact' declaration and its first usage is 11, but allowed 8.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 3]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+
+        Paragraph headerStudent = new Paragraph("Schüler/Schülerin", FONT_TITLE);
+        Paragraph headerEducation = new Paragraph("Schulbildung", FONT_TITLE);
+        Paragraph headerFirm = new Paragraph("Ausbildungsbetrieb", FONT_TITLE);
+        Paragraph headerContact = new Paragraph("Gesetzlicher Vertreter / Nächster Angehöriger", FONT_TITLE);
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:110]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'headerContact' declaration and its first usage is 10, but allowed 8.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 4]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        add(headerFirm);
+        add(tableFirm);
+        add(headerContact);
+        add(tableContact);
+    }
+
+    /**
+     * Used to fill the table with contact information of the enrolled student.
+     * 
+     * @return The contact table.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:131]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 5]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        addTableCell(tableContact, "Telefon (dienstlich): ", FONT_NORMAL);
+        addTableCell(tableContact, enrollmentForm.getRepresentativePhoneNumberCompany(), FONT_NORMAL);
+
+        return tableContact;
+    }
+
+    /**
+     * Used to fill the table with firm information of the enrolled student.
+     * 
+     * @return The firm table.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:175]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 6]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        addTableCell(tableFirm, "E-Mail: ", FONT_NORMAL);
+        addTableCell(tableFirm, enrollmentForm.getEmailCompany(), FONT_NORMAL);
+
+        return tableFirm;
+    }
+
+    /**
+     * Used to fill the table with education information of the enrolled student.
+     * 
+     * @return The education table.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:205]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 7]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        addTableCell(tableEducation, "Sonstige Abschlüsse: ", FONT_NORMAL);
+        addTableCell(tableEducation, enrollmentForm.getDegreesAdditional(), FONT_NORMAL);
+
+        return tableEducation;
+    }
+
+    /**
+     * Used to fill the table with information of the enrolled student.
+     * 
+     * @return The student table.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:227]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 8]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+     */
+
+    private PdfPTable createStudentTable() {
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:230]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.MethodLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Method createStudentTable length is 63 lines (max allowed is 60).

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 9]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+        addTableCell(table, enrollmentForm.getNationality(), FONT_NORMAL);
+        addTableCell(table, "Familiensprache 2: ", FONT_NORMAL);
+        addTableCell(table, enrollmentForm.getFamilyLanguageAdditional(), FONT_NORMAL);
+        addTableCell(table, "Staatsangeh. 2: ", FONT_NORMAL);
+        addTableCell(table, enrollmentForm.getNationalityAdditional(), FONT_NORMAL);
+        addTableCell(table, "Bei Umzug in den letzten 3 Jahren: alte Anschrift: ", FONT_NORMAL);
+
+        AddressEntity addressOld = enrollmentForm.getAddressOld();
+        if (addressOld != null) {
+            addTableCell(table, addressOld.getZip() + ", " + addressOld.getCity() + ", " +
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:283]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'+' should be on a new line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 10]
```diff
@@ -73,25 +84,226 @@ protected String getDocumentTitle() {
...
+            addTableCell(table, "", FONT_NORMAL);
+        }
+
+        return table;
+    }
+
+    /**
+     * Used to fill the paragraph with enrollment information.
+     * 
+     * @return The enrollment paragraph.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:297]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 11]
```diff
@@ -102,5 +314,41 @@ public void initElements() throws DocumentException {
...
+        cell.setBorder(Rectangle.NO_BORDER);
+        cell.setPadding(5);
+
+        table.addCell(cell);
+    }
+
+    /**
+     * Used to translate the gender from the enum into german.
+     * 
+     * @param gender The gender of the enrolled student.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:337]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 12]
```diff
@@ -1,21 +1,30 @@
...
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 
 /**
- * This pdf document retrieves enrollment data for a specific school class and provides a concatinated version of all enrollment forms in the class.
+ * This pdf document retrieves enrollment data for a specific school class and
+ * provides a concatinated version of all enrollment forms in the class.
```
Comment (thisami) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:27]: Du hast den Formatter noch drin 😆

**Review by thisami — APPROVED** [review]

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 13]
```diff
@@ -68,39 +79,246 @@ protected String getDocumentTitle() {
...
      */
     private LocalDateTime fetchEnrollmentDate() {
         return configService.getConfig().getDateEnrollment();
     }
-    
+
+    /**
+     * Adds a cell to the table.
+     * 
+     * @param table The table to add the cell to.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:311]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/data/Gender.java [hunk 14]
```diff
@@ -31,4 +33,17 @@ public String toString() {
         return this.name();
     }
 
+    public String getTitle() {
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/data/Gender.java:36]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/com/fes/hazelnut/schoolnut/data/Gender.java [hunk 15]
```diff
@@ -31,4 +33,22 @@ public String toString() {
         return this.name();
     }
 
+    /**
+     * Returns the title in german.
+     * 
+     * @return The german name of the enum constant.
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/data/Gender.java:39]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 16]
```diff
@@ -73,25 +84,216 @@ protected String getDocumentTitle() {
...
      *
-     * @todo Befüllen.
      */
     @Override
     public void initElements() throws DocumentException {
-        Paragraph paragraphEnrollmentDate = new Paragraph("Die Einschulung findet am " + enrollmentDate.format(dateFormatter) + " um " + enrollmentDate.format(timeFormatter) + " Uhr statt.", FONT_TITLE);
-        
-        paragraphEnrollmentDate.setAlignment(Paragraph.ALIGN_CENTER);
 
+        PdfPTable tableSpacing = new PdfPTable(1);
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:101]: spacing tables als hilfestellung brauchst du nicht, du kannst stattdessen setSpacingBefore und setSpacingAfter bei den anderen Elementen nutzen.

Schätze du hast das vom title in der document vorlage abgeguckt, da ging es nicht anders weil spacing before und after nicht für das aller erste dokument auf der seite funktionieren

**Review by tilokowalski — CHANGES_REQUESTED** [review]
Zum Formatting thema siehe Whatsapp, ansonsten nur ein paar kleinigkeiten

File: src/main/java/com/fes/hazelnut/schoolnut/data/entity/address/AddressEntity.java [hunk 17]
```diff
@@ -48,28 +49,34 @@ public class AddressEntity extends EntityMutuals {
      */
     @OneToOne(targetEntity = EnrollmentFormEntity.class, mappedBy = "addressRepresentative")
     private EnrollmentFormEntity enrollmentFormWhereAddressRepresentative;
-    
+
+     /**
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/data/entity/address/AddressEntity.java:53]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.CommentsIndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Block comment has incorrect indentation level 5, expected is 4, indentation should be the same level as line 56.

**Review by VogtJoshua — APPROVED** [review]

**Review by thisami — DISMISSED** [review]

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java [hunk 18]
```diff
@@ -82,25 +65,204 @@ protected String getFileName() {
...
+
+        return result;
+    }
+
+    /**
+     * Used to fill the table with information of the enrolled student.
+     *
+     * @return The student table.
+     */
+    private PdfPTable createSectionStudent() {
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/EnrollmentFormDocument.java:110]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.MethodLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Method createSectionStudent length is 67 lines (max allowed is 60).

File: src/main/java/com/fes/hazelnut/schoolnut/view/page/EnrollmentPage.java [hunk 19]
```diff
@@ -16,6 +17,8 @@
 import com.vaadin.flow.theme.lumo.LumoUtility;
 import lombok.extern.slf4j.Slf4j;
 
+import lombok.extern.slf4j.Slf4j;
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/view/page/EnrollmentPage.java:20]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'lombok.extern.slf4j.Slf4j'

**Review by tilokowalski — APPROVED** [review]

**Review by VogtJoshua — DISMISSED** [review]

**Review by thisami — APPROVED** [review]

**Pull request merged**